client/src/app-context.tsx

```diff
@@ -1,33 +1,83 @@
-import { useState, createContext, ReactNode, FC, useMemo, useCallback } from 'react'
+import React, { createContext, useContext, useState } from 'react';
+import { ImageHistory } from './editor/image-history';
 
-export interface AppState {
-    imageData: ImageData | null,
-    updateAppState: (newState: Partial<AppState>) => void
-}
+type AppContextType = {
+  imageHistory: ImageHistory;
+  addImageToHistory: (imageData: ImageData) => void;
+  getLastImage: () => ImageData | null;
+  uploadImageToServer: () => Promise<void>;
+};
```

Comment:
Let's not change AppState. We use AppState only to propagate an uploaded image from ImageUploader to Editor, and it shouldn't perform any changes to the images, because Editor should do that.

You upscale image right after it was uploaded, but what we need is to create buttons in the editor after clicking on which the image will be upscaled (it will look like any other filter in the side panel).

I suggest doing this:
- Remove all changes from app-context.tsx
- Create ./src/editor/filters/upscaling.tsx file. Take a look at laplacian-filter.tsx. We need exactly the same architecture for the upscaling filter. 
- Replace code in applyFilter to uploading image to the server
- Add upscaling filter to this line: https://github.com/Tucchhaa/image-filter-editor/blob/main/client/src/editor/editor.tsx#L15. It will create an item in the filters panel